python/dgl/graphbolt/dataloader.py

```diff
@@ -122,7 +187,35 @@ def __init__(self, datapipe, num_workers=0, persistent_workers=True):
             persistent_workers=persistent_workers,
         )
 
-        # (3) Cut datapipe at CopyTo and wrap with prefetcher. This enables the
+        # (3) Overlap UVA feature fetching by buffering and using an alternative
+        # stream.
```

Comment:
why do we need a separate stream?

Reply:
Without seperate stream, overlap is not possible. Need at least 2 streams of execution to overlap them.